datafusion/core/src/dataframe.rs

```diff
@@ -1039,6 +1088,33 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn join_on() -> Result<()> {
+        let left = test_table_with_name("a")
+            .await?
+            .select_columns(&["c1", "c2"])?;
+        let right = test_table_with_name("b")
+            .await?
+            .select_columns(&["c1", "c2"])?;
+        let join = left.join_on(
+            right,
+            JoinType::Inner,
+            [
+                col("a.c1").not_eq(col("b.c1")),
+                col("a.c2").not_eq(col("b.c2")),
```

Comment:
Would it be possible here to also add an equality predicate to demonstrate they are automatically recognized as equi preds?

Perhaps something like

```suggestion
                col("a.c2").eq(col("b.c2")),
```